src/caret_analyze/infra/lttng/ros2_tracing/data_model_service.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2021 Research Institute of Systems Planning, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Set
+
+import pandas as pd
+
+from .data_model import Ros2DataModel
+
+
+class DataModelService:
+
+    def __init__(self, data: Ros2DataModel) -> None:
+        self._data = data
+
+    def get_node_names(self, cbg_addr: int) -> List[str]:
+        """
+        Get node names from callback group address.
+
+        Parameters
+        ----------
+        cbg_addr : int
+            callback group address.
+
+        Returns
+        -------
+        List[str]
+
+        Notes
+        -----
+        Since there may be duplicate addresses in the trace data,
+        this API returns a list of all possible node names.
+
+        """
+        node_names: Set[str] = set()
+        try:
+            node_names |= set(self._get_node_names_from_cbg_timer(cbg_addr))
+        except KeyError:
+            pass
+
+        try:
+            node_names |= set(self._get_node_names_from_cbg_sub(cbg_addr))
+        except KeyError:
+            pass
+
+        try:
+            node_names |= set(
+                self._get_node_names_from_get_parameters_srv(cbg_addr))
+        except KeyError:
+            pass
+
+        if node_names:
+            return sorted(node_names)
+        else:
+            # Failed to identify node name from callback group address.
+            return []
```

Comment:
It seems that this implementation is redundant.
```suggestion
        return sorted(node_names)
```

Reply:
@hsgwa 
Fixed: https://github.com/tier4/CARET_analyze/pull/144/commits/0e46c6286fbdb0c216007aa3eeb71f7c08c46a38
:bow: